tests/test_optim.py

```diff
@@ -27,6 +25,202 @@ def rosen(x):
+            'do_grad-scipy-tensorflow-32b': [0.4999384582042694, 1.0001084804534912],
+            'do_grad-scipy-jax-32b': [0.4999389052391052, 1.0001085996627808],
+            # do grad, scipy, 64b
+            'do_grad-scipy-pytorch-64b': [0.49998837853531425, 0.9999696648069287],
+            'do_grad-scipy-tensorflow-64b': [0.4999883785353142, 0.9999696648069278],
+            'do_grad-scipy-jax-64b': [0.49998837853531414, 0.9999696648069285],
+            # no grad, minuit, 32b - not very consistent for pytorch
+            'no_grad-minuit-numpy-32b': [0.49622172117233276, 1.0007264614105225],
+            #    nb: macos gives different numerics than CI
+            #'no_grad-minuit-pytorch-32b': [0.7465415000915527, 0.8796938061714172],
+            'no_grad-minuit-pytorch-32b': [0.9684963226318359, 0.9171305894851685],
+            'no_grad-minuit-tensorflow-32b': [0.5284154415130615, 0.9911751747131348],
+            #'no_grad-minuit-jax-32b': [0.5144518613815308, 0.9927923679351807],
+            'no_grad-minuit-jax-32b': [0.49620240926742554, 1.0018986463546753],
+            # no grad, minuit, 64b - quite consistent
+            'no_grad-minuit-numpy-64b': [0.5000493563629738, 1.0000043833598724],
+            'no_grad-minuit-pytorch-64b': [0.5000493563758468, 1.0000043833508256],
+            'no_grad-minuit-tensorflow-64b': [0.5000493563645547, 1.0000043833598657],
+            'no_grad-minuit-jax-64b': [0.5000493563528641, 1.0000043833614634],
+            # do grad, minuit, 32b
+            'do_grad-minuit-pytorch-32b': [0.5017611384391785, 0.9997190237045288],
+            'do_grad-minuit-tensorflow-32b': [0.501288652420044, 1.0000219345092773],
+            #'do_grad-minuit-jax-32b': [0.5029529333114624, 0.9991086721420288],
+            'do_grad-minuit-jax-32b': [0.5007095336914062, 0.9999282360076904],
+            # do grad, minuit, 64b
+            'do_grad-minuit-pytorch-64b': [0.500273961181471, 0.9996310135736226],
+            'do_grad-minuit-tensorflow-64b': [0.500273961167223, 0.9996310135864218],
+            'do_grad-minuit-jax-64b': [0.5002739611532436, 0.9996310135970794],
+        }[identifier]
+
+        result = pyhf.infer.mle.fit(data, m, do_grad=do_grad, do_stitch=do_stitch)
+
+        rtol = 2e-06
+        # handle cases where macos and ubuntu provide very different results numerical
+        if 'no_grad-minuit-tensorflow-32b' in identifier:
+            # not a very large difference, so we bump the relative difference down
+            rtol = 3e-02
+        if 'no_grad-minuit-pytorch-32b' in identifier:
+            # quite a large difference
+            rtol = 3e-01
+        if 'do_grad-minuit-pytorch-32b' in identifier:
+            # a small difference
+            rtol = 7e-05
+        if 'no_grad-minuit-jax-32b' in identifier:
+            rtol = 4e-02
+        if 'do_grad-minuit-jax-32b' in identifier:
+            rtol = 5e-03
```

Comment:
My guess is no, but do we have any idea why this is happening for macOS?

Reply:
🤷‍♂️ it's 32b. I'm not super concerned right now, but there's probably reasons why (e.g. processor or 32b / 64b not handled).

Reply:
Okay that's fine. I'll just open an Issue and we can move on from here. :+1:

Reply:
This is PR #981